### PR TITLE
DUPLO-26676 TF: Availability Set: Unable to re-create resource in case of update and forces replacement. Getting error - received unexpected response

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Added retry on post api for inconsistent error
## Summary of changes

This PR does the following:

- updated sku field name in datasource
- added retry at post api

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✔︎] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-26676 TF: Availability Set: Unable to re-create resource in case of update and forces replacement. Getting error - received unexpected response](https://app.clickup.com/t/8655600/DUPLO-26676)

File: duplocloud/resource_duplo_azure_availability_set.go [hunk 1]
```diff
@@ -174,6 +188,7 @@ func resourceAzureAvailabilitySetDelete(ctx context.Context, d *schema.ResourceD
 		return diag.Errorf("Unable to delete tenant %s azure availablity set '%s': %s", tenantID, name, clientErr)
 
 	}
+	time.Sleep(2 * time.Minute)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_availability_set.go:191]: Why we need sleep.

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_azure_availability_set.go:191]: yes, azure is throwing random error when destroyed and recreate. kept this sleep to make azure cooldown and then create the resource.

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_azure_availability_set.go:192]: I tried to use diag := waitForResourceToBeMissingAfterDelete(ctx, d, "azure availability set", id, func() (interface{}, duplosdk.ClientError) {
        return c.AzureAvailabilitySetGet(tenantID, name)
    })
but getting this error
url: https://qa-azure.duplocloud.net/v3/subscriptions/ae909b07-7288-474b-b6a4-b88943bd9cfd/azure/host/availabilityset/ntest3, status: 400, message: {"Message":"Error retrieving AvailabilitySet ntest3 status with Exception The entity was not found in this Azure location."}
BE should have return 404. show using sleep

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**